backend/src/main/java/com/javahelp/backend/accountdeletion/DeleteResult.java

```diff
@@ -0,0 +1,37 @@
+package com.javahelp.backend.accountdeletion;
+
+import com.javahelp.model.token.Token;
+import com.javahelp.model.user.User;
+
+/**
+ * A class used by a {@link DeleteManager} that encodes a successful account deletion result.
+ */
+class DeleteResult {
+    private final User user;
+    private final String successMessage;
+
+    /**
+     * Constructs a {@link DeleteResult} instance for a successful account deletion.
+     *
+     * @param user: the deleted {@link User}.
+     */
+    protected DeleteResult(User user) {
+        this.user = user;
+        successMessage = "Account deletion successful";
```

Comment:
rather than a success message, it might make more sense to have an error message. You could have two constructors, one that takes in an error message for failures, and one that takes in a `User` for successes. You could also add a getter like `isSuccess` or something that returns whether the error message is null.

Reply:
I thought about this, but I wrote it this way because the deletion is made under the circumstance that the user is logged in and there shouldn't be any error in this process. The only error I could think of is if the password doesn't match but since that is tackled by the login use case, I think we will be fine here.

Reply:
That's true, but we also want to be able to inform the end user in cases where something goes wrong with the database, or the string id sent doesn't exist in the first place, or anything else goes wrong. You can also imagine some scenarios where a user is logged in on two instances of the frontend, and deletes their account on one. They may then try to delete their account from the second instance. In this case, we may want to either return an error, or respond with a cached response based on some idempotency key provided with the request. If we do want to return an error, we need a way to do so.